### PR TITLE
cli: use tty to handle password

### LINKED ISSUE #2430: password are handled unsecurely
The same as https://github.com/nspcc-dev/neofs-node/issues/1297
Passwords should be handled directly via /dev/tty, bypassing standard inputs
https://lists.gnupg.org/pipermail/gnupg-devel/2017-August/032987.html

```
% echo qwe | neo-go wallet change-password -w /tmp/wallet.json
Error reading old password: inappropriate ioctl for device
zsh: done       echo qwe | 
zsh: exit 1     ./offline-neofs/downloads/neo-go wallet change-password -w /tmp/wallet.json
% neo-go --version                                 
neo-go version 0.98.3-pre-68-g3b39c8d6
```

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #2430.

### CODE REVIEW

File: cli/input/input.go [hunk 1]
```diff
@@ -48,16 +49,35 @@ func readLine(trm *term.Terminal, prompt string) (string, error) {
...
+			}
+			defer func() { _ = term.Restore(int(syscall.Stdin), s) }()
+			trm = term.NewTerminal(ReadWriter{os.Stdin, os.Stdout}, prompt)
+
 		}
-		defer func() { _ = term.Restore(int(syscall.Stdin), s) }()
-		trm = term.NewTerminal(ReadWriter{os.Stdin, os.Stdout}, prompt)
+		return trm.ReadPassword(prompt)
+	}
+	f, err := os.OpenFile("/dev/tty", os.O_RDWR, 0)
```
Comment (AnnaShaleva) [inline cli/input/input.go:65]: @roman-khimov, how can we test this code?

File: cli/input/input.go [hunk 2]
```diff
@@ -48,16 +49,35 @@ func readLine(trm *term.Terminal, prompt string) (string, error) {
 
 // ReadPassword reads the user's password with prompt.
 func ReadPassword(prompt string) (string, error) {
-	trm := Terminal
-	if trm == nil {
-		s, err := term.MakeRaw(int(syscall.Stdin))
-		if err != nil {
-			return "", err
+	if runtime.GOOS == "windows" {
```
Comment (roman-khimov) [inline cli/input/input.go:52]: I think we can decide on it during compilation.

File: cli/input/input.go (hunk 1 above)
Comment (roman-khimov) [reply cli/input/input.go:65]: You mean our regular tests? Create a pseudo-tty, for example.

**Comment by roman-khimov** [comment]
Does `sshpass` work with the updated code?

File: cli/input/input.go (hunk 1 above)
Comment (AnnaShaleva) [reply cli/input/input.go:65]: Could you, please, clarify, how can I create a pseudo-tty for this case?

Comment (roman-khimov) [reply cli/input/input.go:65]: Something like https://pkg.go.dev/github.com/creack/pty ?

Comment (AnnaShaleva) [reply cli/input/input.go:64]: Ugh, I'm stuck with it, I just can't make our tests work with it, so the current implementation is the best thing I can do for our tests. sshpass works with it as follows:
```
anna@kiwi:~/Documents/GitProjects/nspcc-dev/neo-go$ sshpass -p q ./bin/neo-go contract deploy -i 1-print/1-print.nef -manifest 1-print/1-print.manifest.json -r http://seed1.neo.org:10332 -w wallet.json
Network fee: 0.00151452
System fee: 10.00104553
Total fee: 10.00256005
Relay transaction (y|N)> y
failed to push invocation tx: failed to send tx: InsufficientFunds (-500) - 
```

File: cli/input/input.go (hunk 2 above)
Comment (AnnaShaleva) [reply cli/input/input.go]: Should we create two separate files for UNIX and windows?

Comment (roman-khimov) [reply cli/input/input.go]: Move one function to it, that's perfectly fine.

**Comment by AnnaShaleva** [comment]
However, this implementation doesn't work with `echo` as described in https://github.com/nspcc-dev/neo-go/issues/2387#issue-1160326548.

**Comment by roman-khimov** [comment]
> However, this implementation doesn't work with `echo` as described in [#2387 (comment)](https://github.com/nspcc-dev/neo-go/issues/2387#issue-1160326548).

It shouldn't, that's a feature.

File: cli/input/readpass_unix.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+//go:build !windows
+// +build !windows
```
Comment (fyrchik) [inline cli/input/readpass_unix.go:2]: Starting from `go1.17` the first version is preferred, could you add an issue for that?

**Review by roman-khimov — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]

File: cli/input/readpass_unix.go (hunk 3 above)
Comment (AnnaShaleva) [reply cli/input/readpass_unix.go:2]: Sure, https://github.com/nspcc-dev/neo-go/issues/2497.

**Comment by AnnaShaleva** [comment]
@roman-khimov, then close #2387 also?

**Pull request merged**